identity/validator.go

```diff
@@ -35,9 +41,16 @@ func (v *Validator) Validate(i *Identity) error {
 		NewValidationExtensionIdentifier().WithIdentity(i),
 	}
 
-	err := v.v.Validate(
+	v.d.SchemaPersister()
+
+	s, err := v.d.SchemaPersister().GetSchema(i.TraitsSchemaID)
+	if err != nil {
+		return err
+	}
+
+	err = v.v.Validate(
 		stringsx.Coalesce(
-			i.TraitsSchemaURL,
+			s.URL,
 			v.c.DefaultIdentityTraitsSchemaURL().String(),
```

Comment:
There is no need for this any more because TraitsSchemaID is always set. If it is not set, use to the default _from the database_.